express-server/controllers/cart.controller.js

```diff
@@ -18,15 +18,23 @@ exports.getListShoppingCard = (req, res) => {
 //add to shopping card product
 exports.addProductToShoppingCart = (req, res) => {
     const productId = req.body.productId;
-    const quantity = req.body.quantity;
 
     Buyer.findById(req.user._id)
         .then((buyer) => {
-            buyer.shoppingCart.push({
-                productId: productId,
-                quantity: quantity,
-            });
-
+            let productExist= false;
+            for(let i = 0; i< buyer.shoppingCart.length; i++){
+                if(buyer.shoppingCart[i].productId == productId){
```

Comment:
Did you check it?
I think it will be better if you write like this:
if(buyer.shoppingCart[i].productId**.toString() =**== productId){

because 
buyer.shoppingCart[i].productId - ObjectId
productId - string